File: wpilibjExamples/src/main/java/edu/wpi/first/wpilibj/examples/hatchbottraditional/commands/DriveDistance.java
// Copyright (c) FIRST and other WPILib contributors.
// Open Source Software; you can modify and/or share it under the terms of
// the WPILib BSD license file in the root directory of this project.

package edu.wpi.first.wpilibj.examples.hatchbottraditional.commands;

import edu.wpi.first.wpilibj.examples.hatchbottraditional.subsystems.DriveSubsystem;
import edu.wpi.first.wpilibj2.command.CommandBase;

public class DriveDistance extends CommandBase {
  private final DriveSubsystem m_drive;
  private final double m_distance;
  private final double m_speed;

  /**
   * Creates a new DriveDistance.
   *
   * @param inches The number of inches the robot will drive
   * @param speed The speed at which the robot will drive
   * @param drive The drive subsystem on which this command will run
   */
  public DriveDistance(double inches, double speed, DriveSubsystem drive) {
    m_distance = inches;
    m_speed = speed;
    m_drive = drive;
    addRequirements(m_drive);
  }

  @Override
  public void initialize() {
    m_drive.resetEncoders();
    m_drive.arcadeDrive(m_speed, 0);
  }

  @Override
  public void execute() {
    m_drive.arcadeDrive(m_speed, 0);
  }

  @Override
  public void end(boolean interrupted) {
    m_drive.arcadeDrive(0, 0);
  }

  @Override
  public boolean isFinished() {
    return Math.abs(m_drive.getAverageEncoderDistance()) >= m_distance;
  }
}

File: robotpyExamples/examples/HatchbotTraditional/commands/drivedistance.py
#
# Copyright (c) FIRST and other WPILib contributors.
# Open Source Software; you can modify and/or share it under the terms of
# the WPILib BSD license file in the root directory of this project.
#

import commands2

from subsystems.drivesubsystem import DriveSubsystem


class DriveDistance(commands2.Command):
    def __init__(self, inches: float, velocity: float, drive: DriveSubsystem) -> None:
        self.distance = inches
        self.velocity = velocity
        self.drive = drive
        self.add_requirements(drive)

    def initialize(self) -> None:
        self.drive.reset_encoders()
        self.drive.arcade_drive(self.velocity, 0)

    def execute(self) -> None:
        self.drive.arcade_drive(self.velocity, 0)

    def end(self, interrupted: bool) -> None:
        self.drive.arcade_drive(0, 0)

    def is_finished(self) -> bool:
        return abs(self.drive.get_average_encoder_distance()) >= self.distance
